src/optionals/system.c

```diff
@@ -162,6 +162,36 @@ static Value unameNative(DictuVM *vm, int argCount, Value *args) {
     
     return OBJ_VAL(unameDict);
 }
+
+static Value mkdirTempNative(DictuVM *vm, int argCount, Value *args) {
+    if (argCount == 0) {
+        runtimeError(vm, "mkdirTemp() takes 1 argument (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    if (!IS_STRING(args[0])) {
+        runtimeError(vm, "mkdirTemp() first argument must be a string");
+        return EMPTY_VAL;
+    }
+
+    char *template = AS_CSTRING(args[0]);
+    char *tmpl = {0};
+
+    if (template[0] != '\0') {
+        tmpl = ALLOCATE(vm, char, strlen(template));
+        strcpy(tmpl, template);
+    } else {
+        tmpl = ALLOCATE(vm, char, PATH_MAX);
+        strcpy(tmpl, "XXXXXX");
+    }
```

Comment:
We need to allocate for the terminating char too. My thinking here also is that if no template is passed the directory would be made via the 6 digits of XXXXXX template right? Doesn't that mean we only need to allocate for these 6 + \0?

```suggestion
    if (template[0] != '\0') {
        tmpl = ALLOCATE(vm, char, strlen(template) + 1);
        strcpy(tmpl, template);
    } else {
        tmpl = ALLOCATE(vm, char, 7);
        strcpy(tmpl, "XXXXXX");
    }
```

Reply:
We could also probably keep reference to the length rather than calculating again via strlen in the takeString call